.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+---
+
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+      - feature/github-actions
+  pull_request:
+    branches:
+      - main
+
+# Limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
FYI @v1v prefers to not cancel running builds on main.

See https://github.com/elastic/apm-aws-lambda/pull/370#discussion_r1090825824